src/app/components/challenge/challengesubmit/challengesubmit.component.ts

```diff
@@ -396,15 +389,19 @@ export class ChallengesubmitComponent implements OnInit {
     const submissionFile = self.globalService.formItemForLabel(self.components, 'input_file').fileValue;
     const submissionProjectUrl = self.globalService.formValueForLabel(self.components, 'project_url');
     const submissionPublicationUrl = self.globalService.formValueForLabel(self.components, 'publication_url');
-    const regex = new RegExp('([0-9A-Za-z-\\.@:%_\+~#=]+)+((\\.[a-zA-Z]{2,3})+)(/(.)*)?(\\?(.)*)?');
-    if (submissionFile === null || submissionFile === '') {
+    const submissionFileUrl = self.globalService.formItemForLabel(self.components, 'file_url');
+    const regex = new RegExp(/(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-/]))?/);
+    if (!self.isSubmissionUsingUrl && (submissionFile === null || submissionFile === '')) {
       self.submissionError = 'Please upload file!';
       return;
+    } else if (self.isSubmissionUsingUrl && (submissionFileUrl !== '' && !self.validFileUrl)) {
+      self.submissionError = 'Please enter a valid URL!';
+      return;
     } else if (self.selectedPhase['id'] === undefined) {
       self.submissionError = 'Please select phase!';
       return;
     } else if (submissionProjectUrl !== '' && !regex.test(submissionProjectUrl)) {
-      self.submissionError = 'Please provide a valid project url!';
+      self.submissionError = 'Please enter a valid Submission URL!';
```

Comment:
Hey, this is for project url, please revert this change. You need to modify this message [here](https://github.com/Cloud-CV/EvalAI-ngx/pull/242/files#diff-7ebebddad833dbfe4a95492011383387R398)